provider/cmd/leaseEvents.go

```diff
@@ -36,54 +41,88 @@ func doLeaseEvents(cmd *cobra.Command) error {
 		return err
 	}
 
-	prov, err := providerFromFlags(cmd.Flags())
+	cert, err := cutils.LoadAndQueryCertificateForAccount(cmd.Context(), cctx, cctx.Keyring)
 	if err != nil {
 		return err
 	}
 
-	bid, err := mcli.BidIDFromFlagsForOwner(cmd.Flags(), cctx.FromAddress)
+	dseq, err := dseqFromFlags(cmd.Flags())
 	if err != nil {
 		return err
 	}
 
-	svcs, err := cmd.Flags().GetString(FlagService)
+	leases, err := leasesForDeployment(cmd.Context(), cctx, cmd.Flags(), dtypes.DeploymentID{
+		Owner: cctx.GetFromAddress().String(),
+		DSeq:  dseq,
+	})
 	if err != nil {
 		return err
 	}
 
-	follow, err := cmd.Flags().GetBool("follow")
+	svcs, err := cmd.Flags().GetString(FlagService)
 	if err != nil {
 		return err
 	}
 
-	cert, err := cutils.LoadAndQueryCertificateForAccount(cmd.Context(), cctx, cctx.Keyring)
+	follow, err := cmd.Flags().GetBool("follow")
 	if err != nil {
 		return err
 	}
 
-	gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), prov, []tls.Certificate{cert})
-	if err != nil {
-		return err
-	}
+	ctx, cancel := context.WithCancel(cmd.Context())
+	defer cancel()
 
-	result, err := gclient.LeaseEvents(cmd.Context(), bid.LeaseID(), svcs, follow)
-	if err != nil {
-		return showErrorToUser(err)
+	type result struct {
+		lid      mtypes.LeaseID
+		provider sdk.Address
+		error    error
+		stream   *gwrest.LeaseKubeEvents
 	}
 
-	for res := range result.Stream {
-		if err = cmdcommon.PrintJSON(cctx, res); err != nil {
-			break
+	streams := make([]result, 0, len(leases))
+
+	for _, lid := range leases {
+		var stream result
+		prov, _ := sdk.AccAddressFromBech32(lid.Provider)
+		gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), prov, []tls.Certificate{cert})
+		if err == nil {
+			stream.stream, stream.error = gclient.LeaseEvents(ctx, lid, svcs, follow)
+			stream.lid = lid
```

Comment:
This could be set in an initializer up at the top of the loop like `stream := result{ lid: lid }` so that it is always present in the resulting slice.

Reply:
💯